### PR TITLE
Integrate the Dropbox Exporter

### PR DESCRIPTION
**Pull request by rievo**
With the implementation as is, the user can export to a home-based Dropbox URL (e.g. https://www.dropbox.com/home/testfolder). For the moment, it is not possible to export towards a shared link

### CODE REVIEW

File: app/exporters/dropbox.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                            )
+                            cursor.offset = f.tell()
+
+                f.close()
+
+        except OSError as e:
+            print("Error opening the local file")
+            raise e
+
+        except ApiError:
```
Comment (squeaky-pl) [inline app/exporters/dropbox.py:147]:
```suggestion
        except ApiError as e:
```

File: app/exporters/dropbox.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            print("Error uploading the file")
+            raise FileUploadFailed(f"Error uploading file: {local_path}")
```
Comment (squeaky-pl) [inline app/exporters/dropbox.py:149]:
```suggestion
            raise FileUploadFailed(f"Error uploading file: {local_path}") from e
```

File: app/exporters/dropbox.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        except ValidationError:
```
Comment (squeaky-pl) [inline app/exporters/dropbox.py:151]:
```suggestion
        except ValidationError as e:
```

File: app/exporters/dropbox.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            raise MalformedDropboxURL(
+                "Maybe you are trying to import from a shared link?"
+            )
```
Comment (squeaky-pl) [inline app/exporters/dropbox.py:154]:
```suggestion
            ) from e
```

**Review by squeaky-pl — CHANGES_REQUESTED** [review]

File: app/exporters/dropbox.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+        except ApiError as e:
+            print("Error uploading the file")
+            raise FileUploadFailed(f"Error uploading file: {local_path}") from e
+
+        except ValidationError as e:
+            raise MalformedDropboxURL(
+                "Maybe you are trying to import from a shared link?"
+            ) from e
+            return
```
Comment (squeaky-pl) [inline app/exporters/dropbox.py:153]: This is unreachable code
```suggestion
         
```

**Review by squeaky-pl — CHANGES_REQUESTED** [review]

**Comment by drewler** [comment]
I'll look into why the shared links don't work correcly. Maybe that's just how it works and there's no fix, but I'll take a last look either way.

If everything else works, we'll probably merge anyway as this could be dealt with on a separate PR.

**Review by squeaky-pl — APPROVED** [review]

**Review by drewler — APPROVED** [review]
Looks good to me. Let's merge this one and I'll try to implement support for "shared link" URL format on a separate PR

**Pull request merged**